### PR TITLE
Class

### PR DESCRIPTION
**Pull request by ravencovington**
{
  "comfortability": 3 or 4,
  "completeness": 5,
}

### CODE REVIEW

File: talkative_robot.rb [hunk 1]
```diff
@@ -1,182 +1,251 @@
...
+	end
+end
 
-friend = "Are we friends? (Yes/No)"
-friend = gets.chomp
-puts "Hi bud" unless friend = "No"
+class User < Person
+  attr_accessor :name, :age, :gender
+  def initialize(args)
+  	super(args)
```
Comment (rgpass) [inline talkative_robot.rb]: Check tabbing.

File: talkative_robot.rb [hunk 2]
```diff
@@ -1,182 +1,251 @@
...
+    @gender = args[:gender]
+  end
+end
 
-puts "What's your name?"
-name = gets.chomp
-if name = "Brian" 
-	puts "Hey! You're my boyfriend!"
-else puts "Nope. You are not my boyfriend!"
+name = User.get_name
```
Comment (rgpass) [inline talkative_robot.rb]: Move commands to the bottom.

File: talkative_robot.rb [hunk 3]
```diff
@@ -1,182 +1,251 @@
...
+user = User.new(name: name)
```
Comment (rgpass) [inline talkative_robot.rb]: This should be `user = User.new(name: name, age: age, gender: gender)`

File: talkative_robot.rb [hunk 4]
```diff
@@ -1,182 +1,251 @@
...
+
+class Author < Person
+end
+
+def get_name
+  puts "What's your name?"
+  gets.chomp
+end
```
Comment (rgpass) [inline talkative_robot.rb]: Can remove this method.

File: talkative_robot.rb [hunk 5]
```diff
@@ -1,182 +1,251 @@
...
+
+class GroceryList 
+  def initialize(user)
```
Comment (rgpass) [inline talkative_robot.rb]: Should be initialized with a text file. We are building onto our previous lab.

File: talkative_robot.rb [hunk 6]
```diff
@@ -1,182 +1,251 @@
...
+  end
 end
 
 def age
```
Comment (rgpass) [inline talkative_robot.rb]: Remove

File: talkative_robot.rb [hunk 7]
```diff
@@ -1,182 +1,251 @@
...
-   	return "You'll be 75 in #{75 - age} years"
+    return "You'll be 75 in #{75 - age} years"
    elsif age == 75 
-   	return "You're 75!"
+    return "You're 75!"
    elsif age > 75 
-   	return "You turned 75 #{ age - 75 } years ago"
+    return "You turned 75 #{ age - 75 } years ago"
    end
 end
```
Comment (rgpass) [inline talkative_robot.rb]: Weird tabbing will cause lots of confusion down the line. A lot of this stuff can be removed as well.

**Comment by rgpass** [comment]
All of the methods that were previously written should be moved into their respective classes. Also, there's effectively no inheritance because `Person` has nothing in it.

File: talkative_robot.rb (hunk 4 above)
Comment (ravencovington) [inline talkative_robot.rb]: Should I remove

+def get_name
+  puts "What's your name?"
+  gets.chomp
+end

@rgpass

Comment (rgpass) [inline talkative_robot.rb]: This whole method can be removed and placed as a class method inside `User`

File: talkative_robot.rb (hunk 6 above)
Comment (ravencovington) [inline talkative_robot.rb]: should I remove 
+name = User.get_name
+user = User.new(name: name)
+
+class Author < Person
+end
+
+def get_name
+  puts "What's your name?"
+  gets.chomp
+end
+
+class GroceryList 
+  def initialize(user)
+  end

I'm confused about which line you want removed @rgpass

Comment (rgpass) [inline talkative_robot.rb]: Any methods that are outside of classes should either be moved into a class or removed.

File: talkative_robot.rb [hunk 8]
```diff
@@ -1,10 +1,159 @@
...
+      return "Are you a great-great grandmother?"
+    end
+  end
+end
+
+class User < Person
+  attr_accessor :name, :age, :gender
+  
+  def initialize(args)
+  	super(args)
```
Comment (rgpass) [inline talkative_robot.rb]: Tabbing

File: talkative_robot.rb [hunk 9]
```diff
@@ -1,10 +1,159 @@
...
+    @name = args[:name]
+    @age  = args[:age]
+    @gender = args[:gender]
+    @user = args[:user]
+  end
+
+  def get_name
```
Comment (rgpass) [inline talkative_robot.rb]: Class methods

File: talkative_robot.rb [hunk 10]
```diff
@@ -1,10 +1,159 @@
...
+  def male?
+    user gender] == "M" || user gender] == "male"
+  end
+
+  def male
+    end
+
+  def male?
+    user gender == "M"
+  end
```
Comment (rgpass) [inline talkative_robot.rb]: Have three `male?` functions... only need one.

File: talkative_robot.rb [hunk 11]
```diff
@@ -1,10 +1,159 @@
...
+    gets.chomp
+  end
+
+  def get_user_age
+    puts "Thank you #{user[:name]} How old are you?"
+    gets.chomp.to_i
+  end
+
+  def male?
+    user gender] == "M" || user gender] == "male"
```
Comment (rgpass) [inline talkative_robot.rb]: Probably just want `gender == "M" || gender == "male"`

File: talkative_robot.rb [hunk 12]
```diff
@@ -1,10 +1,159 @@
...
+    gets.chomp.to_i
+  end
+
+  def get_user_gender
+    puts "What's your gender? (M/F)? "
+    gets.chomp
+  end
+  
+  def current_age(user)
+  milestone = 75
```
Comment (rgpass) [inline talkative_robot.rb]: Tabbing

File: talkative_robot.rb [hunk 13]
```diff
@@ -1,10 +1,159 @@
...
+  
+  def random_grocery(groceries)
+      random_grocery = groceries.sample
+      puts "Don't forget the #{random_grocery}"
+  end
+  
+  def length
+      groceries.length
+  end
+end
```
Comment (rgpass) [inline talkative_robot.rb]: All classes and methods should go at the top if building it in a single file. Typically you want to break it into different files though.

File: talkative_robot.rb [hunk 14]
```diff
@@ -1,182 +1,209 @@
...
+    elsif age > 110 && gender == "M"
+      return "Are you a great-great grandfather?"
+    elsif age < 12 && gender == "F"
+      return "You are a young girl"
+    elsif age > 110 && gender == "F"
+      return "Are you a great-great grandmother?"
+    end
+  end
+
+    def gender_formal_greeting
```
Comment (rgpass) [inline talkative_robot.rb]: Tabbing

File: talkative_robot.rb [hunk 15]
```diff
@@ -1,182 +1,209 @@
...
+  end
+  
+  def length
+      groceries.length
+  end
+end
+
+
+
+# # Commands
```
Comment (rgpass) [inline talkative_robot.rb]: Remove comments when they don't add value.

**Comment by rgpass** [comment]
Not sure why this is closed. Homework is good though, so :+1:

**Pull request closed without merging**